src/api/inline-api.ts

```diff
@@ -178,23 +178,25 @@ export class DataviewInlineApi {
      * Takes a filename and arbitrary input data.
      */
     public view(viewName: string, input: any) {
-        /** This cannot be used on systems without file access (i.e. web and mobile devices). */
-        if (!(this.app.vault.adapter instanceof FileSystemAdapter)) {
-            renderErrorPre( this.container, `Dataview: file system access is not available.` );
+
+        let currentFile = this.app.workspace.getActiveFile();
+        let sourcePath  = currentFile?.path ?? '';
+        
+        let viewPath = `${viewName}/view.js`;
+        let viewFile = this.app.metadataCache.getFirstLinkpathDest( viewPath, sourcePath );
```

Comment:
Does this work with folders? I've never tried it.

Reply:
It does, since we're appending `/view.js` first. So we're getting a folder path but then converting it to a file path before we use it. Same thing further down with `/view.css`. I'll improve the file-not-found error message order to clarify where it's expecting the file to be.